src/main/java/jp/classmethod/aws/gradle/cloudformation/AmazonCloudFormationPluginExtension.java

```diff
@@ -173,6 +192,25 @@ public String findStackParameterValue(List<Parameter> cfnStackParameters, String
 		return param.get().getParameterValue();
 	}
 	
+	public String getStackOutputValue(String key) {
+		return findStackOutputValue(getStackOutputs(), key);
+	}
+	
+	public String getStackOutputValue(String stackName, String key) {
+		return findStackOutputValue(getStackOutputs(stackName), key);
+	}
+	
+	public String findStackOutputValue(List<Output> cfnStackOutputs, String key) {
+		Optional<Output> output = cfnStackOutputs.stream()
+			.filter(p -> p.getOutputKey().equals(key))
+			.findAny();
+		if (output.isPresent() == false) {
```

Comment:
So we saying that findStackParameterValue, findStackOutputValue and findPhysicalResourceId should be changed to use Optional string?  If so, I think someone with better Java chops than me should handle.